lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,148 @@ def post(self, request, *args, **kwargs):
+                'video', 'discussion', 'html', 'chapter', 'sequential', 'vertical'.
+            return_type (list, dict): Optional list or dictionary of block's fields based on 'return_type'.
+
+    **Response example**
+
+        Body consists of the following fields:
+
+        root: (str) The ID of the root node of the requested course block structure.\
+        blocks: (dict) A dictionary or list, based on the value of the
+            "return_type" parameter. Maps block usage IDs to a collection of
+            information about each block. Each block contains the following
+            fields.
+
+        id: (str) The Course's id (Course Run key)
+        name: (str) The course's name
+        number: (str) The course's number
+        org: (str) The course's organisation
+        start: (str) Date the course begins, in ISO 8601 notation
+        start_display: (str) Readably formatted start of the course
+        start_type: (str) Hint describing how `start_display` is set. One of:
+            * `"string"`: manually set by the course author
+            * `"timestamp"`: generated from the `start` timestamp
+            * `"empty"`: no start date is specified
+        end: (str) Date the course ends, in ISO 8601 notation
+        media: (dict) An object that contains named media items. Included here:
+            * course_image: An image to show for the course.  Represented
+              as an object with the following fields:
+                * uri: The location of the image
+        certificate: (dict) Information about the user's earned certificate in the course.
+            Included here:
+                * uri: The location of the user's certificate
+        is_self_paced: (bool) Indicates if the course is self paced
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent or the username was not provided
+        * 401 unauthorized, the provided access token has expired and is no longer valid
+          for an authenticated request.
+        * 403 if a user who does not have permission to masquerade as
+          another user specifies a username other than their own.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    def get_certificate(self, request, course_id):
+        """Returns the information about the user's certificate in the course."""
+        if request.user.is_authenticated:
+            certificate_info = certificate_downloadable_status(request.user, course_id)
+            if certificate_info['is_downloadable']:
+                return {
+                    'url': request.build_absolute_uri(
+                        certificate_info['download_url']
+                    ),
+                }
+        return {}
+
+    @staticmethod
+    def compose_course_info(course_overview):
+        """
+        Method for obtaining additional information about the course.
+
+        Arguments:
+            request - Django request object
```

Comment:
This is incorrect, update to say that it expects a CourseOverviews object as input.

Reply:
Thanks, this is a mistake, should be fixed

Reply:
This function was replaced with serializer as suggested